SwiftReflector/Naming/ScopedNaming.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using Dynamo.CSLang;
+
+#nullable enable
+
+namespace SwiftReflector.Naming
+{
+	public class ScopedNaming
+	{
+		string autoSymPrefix;
+		Stack<HashSet<string>> names = new Stack<HashSet<string>> ();
+
+		public ScopedNaming (string autoSymPrefix = "_symbol")
+		{
+			this.autoSymPrefix = autoSymPrefix;
+			names.Push (new HashSet<string> ());
+		}
+
+		public void EnterScope ()
+		{
+			names.Push (new HashSet<string> (names.Peek ()));
+		}
+
+		public void ExitScope ()
+		{
+			if (names.Count == 0)
+				throw new Exception ("exited top level scope - that's bad");
```

Comment:
Is this visible to the customer? If it is, it should be more informative

Reply:
By being public, it is, in theory, visible to the customer but in practice it will only be used by BTfS (or its descendants).
So yeah, everything could be internal, but it makes testing harder.